test/jdk/java/awt/image/multiresolution/MultiDisplayTest.java

```diff
@@ -21,63 +21,106 @@
  * questions.
  */
 
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.BaseMultiResolutionImage;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
+
+import jdk.test.lib.Platform;
+import jtreg.SkippedException;
 
 /*
-  @test
-  @bug 8142861 8143062 8147016
-  @summary Check if multiresolution image behaves properly
-           on HiDPI + non-HiDPI display pair.
-  @author a.stepanov
-  @library /test/lib
-  @build jdk.test.lib.Platform
-  @run applet/manual=yesno MultiDisplayTest.html
-*/
-
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.image.*;
+ * @test
+ * @bug 8142861 8143062 8147016
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @requires (os.family == "windows" | os.family == "mac")
+ * @summary Check if multiresolution image behaves properly
+ *         on HiDPI + non-HiDPI display pair.
+ * @run main/manual MultiDisplayTest
+ */
 
-import jdk.test.lib.Platform;
+public class MultiDisplayTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             The test requires two-display configuration, where
+
+             - 1st display is operating in HiDPI mode;
+             - 2nd display is non-HiDPI.
+
+             In other cases please simply push "Pass".
```

Comment:
These conditions can be detected. On the other hand, since it's a manual test, the tester could enable or connect a second monitor, so it makes sense to preserve the instructions.

Reply:
Do we need an occasional _Skip_ button? Yet adding such a button to throw `jtreg.SkippedException` will introduce a transitive dependency on the library which contains the `SkippedException`.

Reply:
I'm not suggesting modifying the instructions, I'm bringing up it for a discussion.

Essentially, `SkippedException` is also like passing the test, it's indistinguishable in the report. Only the log of the test run will contain the information that the test was skipped.

From this point of view, it's wiser not to introduce an additional dependency which will affect all the manual tests.

Reply:
Makes sense, thanks. The feature would definitely be nice for circumstances like this. This seems to be the first test this sprint that the `Skip` button would apply to. So, not sure how often it would come up. 

Taking into account your point of having an additional dependency on all of these manual tests, maybe better to not have it for now unless similar circumstances pop up more frequently.